core/src/main/scala/sttp/client4/backend.scala

```diff
@@ -52,6 +52,11 @@ trait SyncBackend extends Backend[Identity] {
   override def monad: MonadError[Identity] = IdMonad
 }
 
+/** A [[GenericBackend]] which is synchronous (side effects are run directly), and supports web sockets. */
+trait WebSocketSyncBackend extends SyncBackend with WebSocketBackend[Identity] {
```

Comment:
I think we should have an override for `send` in `WebSocketRequest` with the simplified signature:

```
def send(backend: WebSocketSyncBackend): Response[T] = backend.send(this)
```